### ISSUE TITLE
Counts are prefixed by C-u when displayed in the status bar

### ISSUE THREAD

**Issue by ripread** [syl20bnr/spacemacs#9741]
#### Description :octocat:
In Vim you can type `123j` to go down 123 lines. Emacs also has counts, you type `C-u 123`. Currently in Spacemacs when you type a partial command like `SPC-h` or `g-` the status bar will display the typed combination. When typing `123` the status bar displays `C-u 123`. Its a small thing but it really bothers me. Is there an easy way to fix it?

#### Reproduction guide :beetle:
- Start Emacs
- type `123`
- look at status bar

*Observed behaviour:* :eyes: :broken_heart:
- `C-u 123-` is displayed

*Expected behaviour:* :heart: :smile:
- `123-` is displayed

#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.3.1
- Spacemacs: 0.200.9
- Spacemacs branch: master (rev. 8e1af145)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: ivy
- Layers:
```elisp
(ivy emacs-lisp git syntax-checking)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GCONF GSETTINGS NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 MODULES

**Comment by vinurs** [comment]
^_^, this is emacs' style, do not think like vim

**Comment by CeleritasCelery** [comment]
The effect this the same, But evil is not vim.

**Comment by Alexander-Miller** [comment]
Its actually more correct this way, the numbers you see do count as a [prefix argument](https://www.gnu.org/software/emacs/manual/html_node/elisp/Syntax-Class-Table.html#Syntax-Class-Table), which you most typically enable with C-u. So commands that behave differently with C-u will show the same behaviour when you first enter 12345 instead.

Though there probably is a way to change that. Youll need to dig into spaceline's segment definitions,.

**Comment by bmag** [comment]
Turns out `prefix-command-echo-keystrokes-functions` controls the display of prefix arguments, and by default only `universal-argument--description` is hooked to it. Theoretically we could change it, but then it still needs to show `C-u 123` in cases where `C-u` was actually pressed.

If you really want to change it:
```elisp
(remove-hook 'prefix-command-echo-keystrokes-functions 'universal-argument--description)
(add-hook 'prefix-command-echo-keystrokes-functions 'whatever)
```

Where `whatever` is a function you need to write. For reference, here's how `universal-argument--description` is implemented in Emacs 25.1:
```elisp
(defun universal-argument--description ()
  (when prefix-arg
    (concat "C-u"
            (pcase prefix-arg
              (`(-) " -")
              (`(,(and (pred integerp) n))
               (let ((str ""))
                 (while (and (> n 4) (= (mod n 4) 0))
                   (setq str (concat str " C-u"))
                   (setq n (/ n 4)))
                 (if (= n 4) str (format " %s" prefix-arg))))
              (_ (format " %s" prefix-arg))))))
```

By the way, the display area is called the "echo-area" (it also doubles as the minibuffer window). The status bar is usually called the "mode-line", and is above the echo-area. (Spaceline is a package that modifies the mode-line)

**Comment by bmag** [comment]
I don't count this as a bug, so I labeled it as "Forum". You could report it as a bug to Evil upstream, but I don't think they'll consider it a bug either.
